ez/ez.go

```diff
@@ -0,0 +1,70 @@
+package ez
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/vimeo/dials"
+	"github.com/vimeo/dials/env"
+	"github.com/vimeo/dials/file"
+	"github.com/vimeo/dials/flag"
+	"github.com/vimeo/dials/sourcewrap"
+	"github.com/vimeo/dials/yaml"
+)
+
+// ConfigWithConfigPath is an interface config struct that supplies a
+// ConfigPath() method to indicate which file to read as the config file once
+// populated.
+type ConfigWithConfigPath interface {
+	ConfigPath() (string, bool)
+}
+
+func fileSource(cfgPath string, watch bool) (dials.Source, error) {
+	if watch {
+		fileSrc, fileErr := file.NewWatchingSource(cfgPath, &yaml.Decoder{})
+		if fileErr != nil {
+			return nil, fmt.Errorf("invalid configuration path %q: %s", cfgPath, fileErr)
+		}
+		return fileSrc, nil
+	}
+	fsrc, fileErr := file.NewSource(cfgPath, &yaml.Decoder{})
+	if fileErr != nil {
+		return nil, fmt.Errorf("invalid configuration path %q: %s", cfgPath, fileErr)
+	}
+	return fsrc, nil
+}
+
+// YAMLConfigFlagEnv takes advantage of the ConfigWithConfigPath cfg.
+func YAMLConfigFlagEnv(ctx context.Context, cfg ConfigWithConfigPath, watch bool) (*dials.Dials, error) {
```

Comment:
Could we pass the decoder as a parameter here (and in `fileSource`) so this can be applied to all JSON, YAML, and TOML? Or should we provide another function called JSONConfigFlagEnv if we want to keep the interface simple?